### PR TITLE
NemoAsrReader to optionally return indices of the entries in the manifest.

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It adds new feature needed to map the outputs of NemoAsrReader with the entry from the manifest it belongs to.

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     *Added a ``read_idxs`` option. When true, the operator outputs the indices of the samples as they appear in the manifest file*
     *Bonus: Skip reading the text (and keep it in memory) when not requested by the user*
 - Affected modules and functionalities:
     *NemoAsrReader*
 - Key points relevant for the review:
     *NA*
 - Validation and testing:
     *Tests added*
 - Documentation (including examples):
     *Op documentation updated*

**REQ IDs**: *[RDNEMO.10]*
**JIRA TASK**: *[DALI-2163]*

### CODE REVIEW

File: dali/test/python/test_operator_readers_nemo_asr.py [hunk 1]
```diff
@@ -257,3 +257,25 @@ def _testimpl_nemo_asr_reader_pad_last_batch(batch_size):
...
+def test_read_idxs(batch_size=10):
+  @pipeline_def(device_id=0, num_threads=4)
+  def nemo_asr_reader_read_idxs(reader_seed=12345):
+    audio, idx = fn.readers.nemo_asr(manifest_filepaths=[nemo_asr_manifest], random_shuffle=True, seed=reader_seed,
+                                     read_sample_rate=False, read_text=False, read_idxs=True)
+    return audio, idx
+  pipe1 = nemo_asr_reader_read_idxs(batch_size=batch_size)
+  pipe1.build()
+  pipe2 = nemo_asr_reader_read_idxs(batch_size=batch_size)
+  pipe2.build()
```
Comment (JanuszL) [inline dali/test/python/test_operator_readers_nemo_asr.py:270]: What is the value in comparing two, identical piepelines?

Comment (jantonguirao) [reply dali/test/python/test_operator_readers_nemo_asr.py:270]: I wanted to check that the order of indices is the same for the same seed

Comment (JanuszL) [reply dali/test/python/test_operator_readers_nemo_asr.py:270]: Ok

Comment (JanuszL) [reply dali/test/python/test_operator_readers_nemo_asr.py:270]: So how about setting the same see in the L267 and L269 explicitly?

Comment (jantonguirao) [reply dali/test/python/test_operator_readers_nemo_asr.py:270]: done

**Review by JanuszL — APPROVED** [review]

File: dali/operators/reader/loader/nemo_asr_loader.h [hunk 2]
```diff
@@ -89,7 +95,8 @@ namespace detail {
 
 DLL_PUBLIC void ParseManifest(std::vector<NemoAsrEntry> &entries, std::istream &manifest_file,
                               double min_duration = kDefaultDuration,
-                              double max_duration = kDefaultDuration);
+                              double max_duration = kDefaultDuration,
+                              bool read_text = true);
```
Comment (szalpal) [inline dali/operators/reader/loader/nemo_asr_loader.h:99]: I know that reading whole reader makes the `read_text` meaning obvious, but when somebody reviews only a part of it, `read_text` looks ambiguous. Could you add a doc to this function?

Comment (jantonguirao) [reply dali/operators/reader/loader/nemo_asr_loader.h:106]: Done

**Comment by jantonguirao** [comment]
!build

**Comment by jantonguirao** [comment]
!build

**Comment by jantonguirao** [comment]
!build

**Review by szalpal — APPROVED** [review]

**Pull request merged**